### PR TITLE
T33043 Update generate-eos-extra script to feature apps in more places

### PR DESCRIPTION
**Pull request by pwithnall**
Various commits to tidy `generate-eos-extra` up, and expand the XML it outputs to show the apps in more featured places in gnome-software.

See the commit messages for details.

https://phabricator.endlessm.com/T33043

### CODE REVIEW

File: app-info/generate-eos-extra [hunk 1]
```diff
@@ -82,13 +82,27 @@ def main():
                 len(appstream_id.split('.')) > 3):
             appstream_id = appstream_id[:-len(".desktop")]
 
+        # Add XML to include the app in the carousel on the overview page, the
+        # “Editor’s Choice” section on the overview page, the carousel on
+        # category pages, and the “Editor’s Choice” section on category pages.
+        #
+        # This will only work for apps which have a hi-res icon (defined in the
+        # main metainfo entry for that app).
```
Comment (wjt) [inline app-info/generate-eos-extra:113]: Interesting, does this apply only to the carousel, or anywhere?

I ask because the SOMA apps only have 64×64px icons and this is unlikely to change.

File: app-info/generate-eos-extra [hunk 2]
```diff
@@ -82,13 +82,27 @@ def main():
...
+        #
+        # See https://gitlab.gnome.org/GNOME/gnome-software/-/blob/main/doc/vendor-customisation.md#user-content-featured-apps-and-editors-choice
```
Comment (wjt) [inline app-info/generate-eos-extra:115]: Perhaps add an XML example to that page, with comments to clarify which kudo/custom/category causes which behaviours.

File: app-info/generate-eos-extra [hunk 3]
```diff
@@ -90,6 +97,12 @@ def main():
                 len(appstream_id.split('.')) > 3):
             appstream_id = appstream_id[:-len(".desktop")]
 
+        # FIXME: Build the existing list of categories for the app, as we cannot
+        # append a new category, only overwrite the whole list of categories.
+        # This is due to a tedious bug in gnome-software:
+        # https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1649
```
Comment (wjt) [inline app-info/generate-eos-extra:105]: :scream_cat:

**Review by wjt — APPROVED** [review]
LGTM.

File: app-info/generate-eos-extra (hunk 1 above)
Comment (pwithnall) [reply app-info/generate-eos-extra:113]: It applies to the carousel and everywhere on the category pages. i.e. It doesn’t apply to editor’s choice on the overview page.

tl;dr: SOMA apps could be put in editor’s choice on the overview page, but not in an individual category. That’s probably OK since if you go into the Education category the SOMA apps are fairly obvious anyway (as they are quite numerous).

File: app-info/generate-eos-extra (hunk 2 above)
Comment (pwithnall) [reply app-info/generate-eos-extra:115]: https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/1315

**Comment by pwithnall** [comment]
Merged as per [Will’s review](https://phabricator.endlessm.com/T33043#929963). I’ve opened https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/1315 upstream with a bit of follow-up.
